src/ocf_core.c

```diff
@@ -251,25 +243,14 @@ void ocf_core_submit_io_mode(struct ocf_io *io, ocf_cache_mode_t cache_mode)
 		return;
 	}
 
+	req->part_id = ocf_part_class2id(cache, io->io_class);
+	req->core = core;
+	req->complete = ocf_req_complete;
+
 	/* TODO: instead of casting ocf_cache_mode_t to ocf_req_cache_mode_t
 	   we can resolve IO interface here and get rid of the latter. */
```

Comment:
Add:
req->cache_mode = cache_mode;

Pls check whether the same change applies to submit_fast.